services/reading-contest-api/interfaces/repositories/contest_repository.go

```diff
@@ -138,3 +138,62 @@ func (r *contestRepository) FindByID(id uint64) (domain.Contest, error) {
 
 	return contest, nil
 }
+
+func (r *contestRepository) Stats(id uint64) (domain.ContestStats, error) {
+	var contestStats domain.ContestStats
```

Comment:
For other features i'd most likely split out individual queries into different functions, but it's ok for this one. I don't see this one being changed all that often in the near future.